### PR TITLE
luarocks: update to 3.0.4.

### PR DESCRIPTION
**Pull request by maxice8**
closes #8887

### CODE REVIEW

**Comment by maxice8** [comment]
Side-effects:

- you now need lua5.3 even if you only want lua5.2 or lua5.1
- There are no alternatives=

File: srcpkgs/luarocks/template [hunk 1]
```diff
@@ -1,38 +1,77 @@
...
-	sed -i -e 's;env lua;env lua5.3;g' src/bin/luarocks{,-admin}
+	vsed -i -e 's;env lua;env lua5.3;g' src/bin/luarocks{,-admin}
 }
+
 post_install() {
 	vlicense COPYING
-	rm -rfv ${DESTDIR}/usr/bin/luarocks{,-admin}
+
+	make DESTDIR="${DESTDIR}" LUA_VERSION=5.1 install-config
+	make DESTDIR="${DESTDIR}" LUA_VERSION=5.2 install-config
```
Comment (Duncaen) [inline srcpkgs/luarocks/template]: I would move this into the for loop, and the other make invocation too.

**Comment by maxice8** [comment]
Ah good suggestions I just copied them from the pkgbuild

File: srcpkgs/luarocks/template [hunk 2]
```diff
@@ -1,38 +1,77 @@
...
 build_style=configure
-configure_args="
- --prefix=/usr
- --sysconfdir=/etc/luarocks
- --lua-version=5.3
- --lua-suffix=5.3
- --with-lua-include=/usr/include/lua5.3
- --with-downloader=curl
+configure_args=" --prefix=/usr --sysconfdir=/etc
+ --lua-version=5.3 --with-lua-include=/usr/include/lua5.3
```
Comment (Duncaen) [inline srcpkgs/luarocks/template]: I prefer the previous style for clarity, but whatever.

File: srcpkgs/luarocks/template (hunk 1 above)
Comment (maxice8) [reply srcpkgs/luarocks/template]: done

File: srcpkgs/luarocks/template (hunk 2 above)
Comment (maxice8) [reply srcpkgs/luarocks/template]: reverted

**Pull request merged**